src/client/pages/myRecommendations/MyRecommendations.jsx

```diff
@@ -0,0 +1,228 @@
+        isbn: '923198dopk0i321',
+        description: 'When mysterious letters start arriving on his doorstep, Harry Potter has never heard of Hogwarts School of Witchcraft and Wizardry. They are swiftly confiscated by his aunt and uncle. Then, on Harry\'s eleventh birthday, a strange man bursts in with some important news: Harry Potter is a wizard and has been awarded a place to study at Hogwarts. And so the first of the Harry Potter adventures is set to begin.',
+        cover: 'https://upload.wikimedia.org/wikipedia/en/6/6b/Harry_Potter_and_the_Philosopher%27s_Stone_Book_Cover.jpg',
+        authors: [
+            'J.K. Rowling'
+        ],
+        genres: [
+            'wizard',
+            'fiction'
+        ],
+        tags: [
+            'stones',
+            'magic',
+            'harry potter',
+            'awesomeness'
+        ]
+    },
+    {
+        id: 3,
+        title: 'The Maze Runner',
+        isbn: '9dsa90udfadwk',
+        description: 'A teen wakes up in a clearing in the center of a gigantic maze with no memory of his past, finding himself a resident in community of boys who have built a village in the glade and who sends two of its strongest and fittest runners into the maze every morning to find a way out.',
+        cover: 'https://toppsta.com/images/covers/9/4/0/0/9781909489400.jpg',
+        authors: [
+            'James Dashner'
+        ],
+        genres: [
+            'spooky',
+            'fiction'
+        ],
+        tags: [
+            'maze',
+            'running'
+        ]
+    }
+]
+
+const recommendedAuthors = [
+    'J.K. Rowling',
+    'J.R.R. Tolkien',
+    'Stephen King',
+    'Margaret Atwood',
+    'Lena Jones',
+    'Agatha Christie',
+    'Jane Austen',
+    'Danielle Steel'
+]
+
+const recommendedTags = [
+    'Stones',
+    'Magic',
+    'Slice of Life',
+    'Cool',
+    'Nice',
+    'Awesomeness',
+    'Violence',
+    'Swearing',
+    'Sweet'
+]
+
+
+
+const MyRecommendations = () => {
+    const sliceStart = 0
+    const sliceEnd = 5
+
+    const [books, setBooks] = useState(recommendedBooks)
+    const [authors, setAuthors] = useState(recommendedAuthors)
+    const [tags, setTags] = useState(recommendedTags)
+    const [displayConfirm, setDisplayConfirm] = useState(false)
+    const [selectedBook, setSelectedBook] = useState(null)
+
+    const removeTag = (e) => {
+        e.preventDefault()
+        const removeItem = e.target.className
+        const removeIndex = tags.indexOf(removeItem)
+        const deleteCount = 1
+        const newTags = [...tags]
+        newTags.splice(removeIndex, deleteCount)
+        setTags(newTags)
+    }
+
+    const removeAuthor = (e) => {
+        e.preventDefault()
+        const removeItem = e.target.className
+        const removeIndex = authors.indexOf(removeItem)
+        const deleteCount = 1
+        const newAuthors = [...authors]
+        newAuthors.splice(removeIndex, deleteCount)
+        setAuthors(newAuthors)
+    }
```

Comment:
These two functions are almost identical which violates one of the core principals of development (Don't Repeat Yourself). I think you could extract the core logic into its own function here so you'd end up with something like the below...

```js
const removeAuthor = (e) => {
  e.preventDefault()
  const newAuthors = remove(authors, e.target.className);
  setAuthors(newAuthors);
}
```

Reply:
I'm not sure what that function does, never seen remove on it own before. Do I have to declare remove myself?

Reply:
He's telling you to create a remove function on your own that performs those instructions with the given parameter. (So you will call it once with authors and once with tags)